phoonnx_train/optispeech/model/optispeech.py

```
@@ -0,0 +1,283 @@
+        """
+        languages = self.text_processor.languages
+        if language is None:
+            language = languages[0]
+        if self.num_speakers > 1:
+            if speaker is None:
+                sid = 0
+            elif type(speaker) is str:
+                try:
+                    sid = self.speakers.index(speaker)
+                except IndexError:
+                    raise ValueError(f"A speaker with the given name `{speaker}` was not found in speaker list")
```

Ruff (0.15.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)